### PR TITLE
Fix vertex representation and adjust camera range

### PR DESCRIPTION
**Pull request by pentacular**
Large numbers were using exponential notation, breaking objects larger than about one meter.

The camera was also focused too close making large structures difficult to see.

Also introduces @preact/signals to make persistent config easier to delegate.

### CODE REVIEW

**Comment by BarbourSmith** [comment]
You are on a roll!! 👍 👍

File: ui/app/SheetStorage.js [hunk 1]
```diff
@@ -233,6 +243,32 @@ export class SheetStorage {
   }
 }
 
+class Config {
+  constructor(path, defaultValue = '') {
+    this.defaultValue = defaultValue;
+    this.path = path;
+    this.signal = signal(defaultValue);
```
Comment (BarbourSmith) [inline ui/app/SheetStorage.js:250]: What do signals do?

File: ui/app/package.json [hunk 2]
```diff
@@ -36,7 +36,8 @@
     "flexlayout-react": "^0.5.20",
     "hash-sum": "^2.0.0",
     "marked": "^4.0.10",
-    "preact": "10.5.14",
+    "preact": "10.25.0",
+    "@preact/signals": "1.3.0",
```
Comment (BarbourSmith) [inline ui/app/package.json:40]: Ah I guess I can answer my own question 😛

**Review by BarbourSmith — APPROVED** [review]
It's exciting to see things moving so quick.

Have you been modeling anything cool lately?

File: ui/app/SheetStorage.js (hunk 1 above)
Comment (pentacular) [reply ui/app/SheetStorage.js:250]: They're variables which trigger re-rendering on change.

They seem like they might be an improvement over passing state all over the place.

I'm hoping that they allow much of the logic to become decentralized.

**Comment by pentacular** [comment]
> It's exciting to see things moving so quick.
> 
> Have you been modeling anything cool lately?

I'm not sure it's cool, but I have been modeling some desk design / room layout.

image

Although that's an old version which doesn't have the room layout in it.

But it did show me problems with large objects. :)

**Comment by BarbourSmith** [comment]
That is looking awesome!

**Comment by BarbourSmith** [comment]
How is packing working these days?

**Pull request merged**